src/middlewares/preInvokeMiddleware.tsx

```diff
@@ -0,0 +1,34 @@
+/**
+ * Process preInvoke operation before action sendOperation
+ */
+
+import {AnyAction, Dispatch, Middleware, MiddlewareAPI} from 'redux'
+import {$do, types} from '../actions/actions'
+import {Store as CoreStore} from '../interfaces/store'
+
+const preInvokeAction = ({getState, dispatch}: MiddlewareAPI<Dispatch<AnyAction>, CoreStore>) => (next: Dispatch) =>
+    (action: AnyAction) => {
+        if (action.type === types.sendOperation) {
+            const bcName = action.payload.bcName
+            const operationType = action.payload.operationType
+            const widgetName = action.payload.widgetName
+            const preInvoke = action.payload?.confirmOperation
```

Comment:
`confirmOperation` field is deprecated:
https://github.com/tesler-platform/tesler-ui/blob/04f4874f83f6072ad938bdb749cb2d8cd5d03899/src/actions/actions.ts#L339-L342

IDE would show it if not for the missing typings; in some cases typescript can deduce the type based on the condition branch but not always: in this case it couldn't so you'll have to use manual case or write a typeguard to get correct typings for payload.

And with correct typings IDE will show that `confirmOperation` field is deprecated.
The reasoning is that when you need a pre-invoke you can calculate it based on `widgetName` and `operationType`: get the widget, read bcName, use bcName to access current row meta, use operationType to get the operation, read the `preInvoke` from there.  
The `confirmOperation` field is optional and intended for cases when you hardcode confirmation logic from front end.

Reply:
fixed. Get `preInvoke` Operation from state